.buildkite/scripts/test.sh

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+echo "--- Prepare enviroment"
+source .buildkite/scripts/pre-install-command.sh
+add_bin_path
+with_go_junit_report
+
+echo "--- Run the tests"
+export OUT_FILE="build/test-report-${GO_VERSION}"
+mkdir -p build
+set +e
+go test -v -race ./... > "${OUT_FILE}"
+status=$?
```

Comment:
Should it be added again `set -e` after getting the exit code ? 

```suggestion
mkdir -p build
set +e
go test -v -race ./... > "${OUT_FILE}"
status=$?

set -e
```